examples/CSVImporter/CSVImporter.go

```diff
@@ -0,0 +1,261 @@
+package main
+
+import (
+    "fmt"
+    "flag"
+    "encoding/csv"
+    "os"
+    "io"
+    "github.com/TheJumpCloud/jcapi"
+)
+
+
+//
+// This program will process each line (record) of a CSV file as a user
+// import request into JumpCloud.
+//
+// The CSV file must have each line formatted as:
+// 
+// first_name, last_name, USER_NAME, EMAIL, uid, gid, SUDO_FLAG, password, host_name, tag_name, admin1, admin2, ...
+//
+// Values shown in all lowercase are optional, while those in all uppercase
+// are required.
+//
+// For each line of the CSV file, this program will:
+//
+// 1 - Insert the USER_NAME specified as a new JumpCloud user, if the name
+//     does not already exist, otherwise this line will be treated as an
+//     update to an existing user matching USER_NAME.
+//
+// 2 - If the password was specified, the USER_NAME will have that password
+//     assigned to it immediately, otherwise, JumpCloud will automatically
+//     generate an email message, delivered to the EMAIL specified, requesting
+//     the user to provide an appropriate password for their account.
+//
+// 3 - The value of the SUDO_FLAG will always be applied to this user.  All
+//     other optional values specified will be applied as appropriate.
+//
+// 4 - When both the host_name and tag_name are specified, a tag will be
+//     created or updated for this USER_NAME associated with the host_name
+//     provided.
+//     (n.b. - if the USER_NAME already exists on the host_name in question,
+//     specifying these values here will result in the account on host_name
+//     being "taken over" by JumpCloud)
+//
+// 5 - Any invalid combination of values will generate an error and terminate
+//     processing for that entry, but the program will attempt to process the
+//     remainder of the file.
+//
+// 6 - A summary is printed at the conclusion of processing.
+//
+
+
+const (
+    urlBase string = "https://console.jumpcloud.com/api"
+)
+
+
+//
+// Returns the ID for the username specified if it is contained in the
+// list of users provided.  (helper function)
+//
+
+func GetUserIdFromUserName(users []jcapi.JCUser, name string) string {
+    returnVal := ""
+
+    for _, user := range users {
+        if user.UserName == name {
+            returnVal = user.Id
+            break
+        }
+    }
+
+    return returnVal
+}
+
+
+//
+// Main Entry Point...
+//
+
+func main() {
+    // Input parameters
+    var apiKey string
+    var csvFile string
+
+	// Obtain the input parameters
+	flag.StringVar(&csvFile, "csv", "", "-csv=<filename>")
```

Comment:
I doubt it.  It look OK on my screen, naturally.  Sidebar with with the correct sublime config stuff and I'll fix that up.